cmd/snap-confine/seccomp-support.c

```diff
@@ -109,12 +135,94 @@ static void validate_bpfpath_is_safe(const char *path)
 	}
 }
 
+static void sc_cleanup_sock_fprog(struct sock_fprog *prog)
+{
+	free(prog->filter);
+	prog->filter = NULL;
+}
+
+static void sc_must_read_filter_from_file(FILE *file, uint32_t len_bytes,
+					  char *what, struct sock_fprog *prog)
+{
+	if (len_bytes == 0) {
+		die("%s filter may only be empty in unrestricted profiles",
+		    what);
+	}
+	prog->len = len_bytes / sizeof(struct sock_filter);
+	prog->filter = (struct sock_filter *)malloc(len_bytes);
```

Comment:
The cast is not needed in C

Reply:
Will do

Reply:
Done in 2ae52cbfa2d8635f7365cbe8e21ad31dd4a02f57